### PR TITLE
Fix sporadic issues in yast2_instserver

### PR DESCRIPTION
**Pull request by hjluo**
- Description:
    -   Fix sporadic issues in yast2_instserver

- Related ticket: [**153379**](https://progress.opensuse.org/issues/153379)

- Related MR:  [**700**](https://gitlab.suse.de/qa-maintenance/qam-openqa-yml/-/merge_requests/700)
- Needles: N/A
- Verification run: 
   - [**VR**](https://openqa.suse.de/tests/overview?build=hjluo%2Fos-autoinst-distri-opensuse%23extratests-yast2u&version=15-SP4&distri=sle) **_20/20 PASSED_**
   -  Fixed
        -  https://openqa.suse.de/tests/13239241#step/yast2_instserver/10
        -  https://openqa.suse.de/tests/13239238#step/yast2_instserver/40
        - https://openqa.suse.de/tests/13239257#step/yast2_instserver/10

### CODE REVIEW

File: lib/y2_module_guitest.pm [hunk 1]
```diff
@@ -60,7 +62,8 @@ sub launch_yast2_module_x11 {
     diag 'assuming root-auth-dialog, typing password';
     type_password;
     send_key 'ret';
-    assert_screen $args{target_match}, $args{match_timeout};
+    wait_still_screen(timeout => 300);
```
Comment (lemon-suse) [inline lib/y2_module_guitest.pm:65]: Does this needed? If the timeout is not enough, I think we can increase the timeout value when run the launch_yast2_module_x11.

Comment (hjluo) [reply lib/y2_module_guitest.pm:65]: > Does this needed? If the timeout is not enough, I think we can increase the timeout value when run the launch_yast2_module_x11.

Sometimes it'll freeze here till timeout https://openqa.suse.de/tests/13239257#step/yast2_instserver/10, and the workaround helps.

File: lib/y2_module_guitest.pm [hunk 2]
```diff
@@ -60,7 +62,8 @@ sub launch_yast2_module_x11 {
     diag 'assuming root-auth-dialog, typing password';
     type_password;
     send_key 'ret';
-    assert_screen $args{target_match}, $args{match_timeout};
+    wait_still_screen 2, 2;
```
Comment (jknphy) [inline lib/y2_module_guitest.pm:65]: this if needed at all, should be only inside the if.

File: lib/y2_module_guitest.pm [hunk 3]
```diff
@@ -10,6 +10,8 @@ use warnings;
 use utils;
 use testapi;
 use Exporter 'import';
+use version_utils "is_sle";
```
Comment (jknphy) [inline lib/y2_module_guitest.pm:13]: single quotes.

Comment (hjluo) [reply lib/y2_module_guitest.pm:13]: changed.

**Review by lemon-suse — APPROVED** [review]

File: lib/y2_module_guitest.pm (hunk 2 above)
Comment (hjluo) [reply lib/y2_module_guitest.pm]: removed.

**Comment by hjluo** [comment]
> Please run VRs with latest code for at least 10 times for this is a sporadic issue.

done, [**extratests-yast2u-mau-extratests-yast2ui_dev1_1**](https://openqa.suse.de/tests/overview?build=hjluo%2Fos-autoinst-distri-opensuse%23extratests-yast2u&version=15-SP4&distri=sle)

**Comment by lemon-suse** [comment]
LGTM.

**Comment by jknphy** [comment]
LGTM, instead of having "MR: TBA" in the ticket I would prefer to have it here in the PR indicated in the first comment (we normally do like this), so the PR and MR can be merged at the same time, otherwise we could forget or not review properly because sometimes there are dependent changes (not this case though).

**Comment by hjluo** [comment]
> LGTM, instead of having "MR: TBA" in the ticket I would prefer to have it here in the PR indicated in the first comment (we normally do like this), so the PR and MR can be merged at the same time, otherwise we could forget or not review properly because sometimes there are dependent changes (not this case though).

Done.

**Pull request merged**